### PR TITLE
Fixed: Better names for nlp variables in pipelines' tests and docs.

### LINKED ISSUE #9455: Rename `nlp` variables into more appropriate names
# 🚀 Feature request

In `pipelines` tests and documentation they are recurringly named `nlp`, the goal is to rename them to something more appropriate.

## Motivation

```python
nlp = pipeline(task='conversational', model='XXX')
```
This is a bit pretentious as it does cover all NLP, and a better name help understanding to users too.
For instance the `conversational` task pipeline could be named `conversational_agent`. Or maybe still a generic name but less
pretentious `pipe`, `pipeline` (caveat: those are less clear in what they intend to achieve)

The goal is to rename all occurences of `nlp` by better names within both tests and the documentation.

## Your contribution

The better names could be discussed here.

### PR DESCRIPTION
**Pull request by 01-vyom**
# What does this PR do?

Fixes #9455 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@Narsil @sgugger

### CODE REVIEW

File: docs/source/task_summary.rst [hunk 1]
```diff
@@ -336,14 +336,14 @@ Here is an example of using pipelines to replace a mask from a sequence:
 
     >>> from transformers import pipeline
 
-    >>> nlp = pipeline("fill-mask")
+    >>> unmask = pipeline("fill-mask")
```
Comment (sgugger) [inline docs/source/task_summary.rst:339]: Could we call it `unmasker` here and in the two instances below?

**Review by sgugger — APPROVED** [review]
Amazing, thanks a lot for doing this!

**Review by LysandreJik — APPROVED** [review]

**Comment by sgugger** [comment]
Could you take care of the merge conflicts and we should be good to merge? Thanks!

**Comment by Narsil** [comment]
Thanks a lot for this !

**Pull request merged**